resharper/src/resharper-unity/CSharp/Daemon/Errors/CSharpErrors.xml

```diff
@@ -52,6 +52,30 @@
         <Description>Explicit string comparison with 'GameObject.tag' or 'Component.tag' is inefficient. Use the 'CompareTag' method instead.</Description>
       </Tag>
 
+      <Tag externalName="UseExplicitTypeInsteadOfStringUsingWarning.HIGHLIGHTING_ID"
+           default="WARNING">
+        <Title>Use type method instead of overload with string</Title>
+        <Description>Using the string for the component type variants results in additional overhead that should not be incurred</Description>
+      </Tag>
+
+      <Tag externalName="InvalidStringForTypeWarning.HIGHLIGHTING_ID"
+           default="WARNING">
+        <Title>Correct type is not found</Title>
+        <Description>The type could not be resolved from string</Description>
+      </Tag>
+
+      <Tag externalName="AmbiguousUnityMonoBehaviourDefinitionWarning.HIGHLIGHTING_ID"
+           default="WARNING">
+        <Title>Inheritor of 'MonoBehaviour' with same name defined twice in different namespaces</Title>
+        <Description>Its dangerous to use name of this type in methods like 'GetComponent' due to unexpected behaviour</Description>
+      </Tag>
+      
```

Comment:
You can probably use the `MultipleResolveCandidatesInTextWarning` here. It'll say:

> Ambiguous reference: 
> Thing.Foo
> Whatever.Foo
> match

Alternatively:

Title: "Ambiguous MonoBehaviour reference: {0} match"
Description: "The string literal overload of methods such as GameObject.AddComponent, GameObject.GetComponent and ScriptableObject.CreateInstance can use non-qualified names, and cause ambiguous references, where one class is selected at random at runtime. Prefer fully qualified names, or the generic method overload"

Also, does this apply to `ScriptableObject.CreateInstance`?